internal/internal_workflow_client.go

```diff
@@ -1180,6 +1186,37 @@ func (wc *WorkflowClient) UpdateWorkflow(
 	return wc.interceptor.UpdateWorkflow(ctx, in)
 }
 
+func (wc *WorkflowClient) UpdateWithStartWorkflow(
+	ctx context.Context,
+	options UpdateWithStartWorkflowOptions,
+) (WorkflowUpdateHandle, error) {
+	startOp, ok := options.StartOperation.(*WithStartWorkflowOperationImpl)
+	if !ok {
+		return nil, errors.New("startOperation must be created by NewWithStartWorkflowOperation")
+	}
+	if startOp.err != nil {
+		return nil, startOp.err
+	}
```

Comment:
This negates the benefit of returning the interface. Can just make it optional at this level.

```suggestion
	if startOp, _ := options.StartOperation.(*WithStartWorkflowOperationImpl); startOp != nil && startOp.err != nil {
		return nil, startOp.err
	}
```

But it is understood that in the root interceptor the raw type is needed

Reply:
If a users is mocking the client then this code is not relevant

Reply:
Technically another benefit of the interface here is that it could be a different impl and be made into the proper impl by an interceptor, but I agree that is far fetched. This code doesn't have to change.